### ISSUE TITLE
Understanding what the parameters/calculations mean

### ISSUE THREAD

**Issue by pmatos** [Bouni/python-luxtronik#35]
Hi,

I have a few questions:

1. What's the difference between parameters and calculations?
2. Do you hold a file somewhere about the meaning of the parameters you know about?

For example, If I open my Alpha Innotec Control application I can see values and explanations, I was wondering if I could print all values available and try to associate them to an explanation of what they are. I could then add to the documentation you already publicly have. What do you think?

Also, is there a way to cycle through all parameters, calculations and print their values?

**Comment by kbabioch** [comment]
I'm familiar with gettext and have used it across some projects (although not necessarily with Python). 

However I'm not yet convinced that "abusing" strings as part of the class is the right way to document the parameters. While something like `Temperatur Vorlauf` is already more descriptive than `ID_WEB_Temperatur_TVL`.

In a perfect world the documentation should also contain more details, i.e. what this parameter means (physically), what the allowed values are (data type, min, max, etc.).

Typically I would suggest to use docstrings for this, so that the documentation lives next to the code to ensure it stays consistent as good as possible.

There are some problems though:

- The essential aspects that need documentation are entries in a dictionary. Not sure how well docstrings will work for that, some initial tests were not too promising.
- No i18n: The code will be documented in English most likely. I'm not familiar with docstrings in multiple languages. For me English is perfectly fine, but some users of the library might be unhappy with that. I'm pretty sure that the user base of this library are owners of AIT products, which is probably widespread in Germany, but nowhere else?

**Comment by Bouni** [comment]
Maybe the docs could live in a seperate file and matched by the id, that way we could have a proper name, description, min, max etc. as well as i18n support (at least thats waht I assume, nver really used it)

Something like 

**docs.py**
```python
paramter_docs = {
  10: {
    "name": _("Vorlauftemperatur"),
    "description": _("bla bla"),
    "min": 10.0,
    "max": 50.0
  }
}
```

`_()` the is used to get the correct translation via `gettext`.

I guess we need to pass the id then into the class to have it available for getting the docs, something like `10: Celsius("ID_WEB_Temperatur_TVL", 10),`

**Comment by BenPru** [comment]
> **docs.py**
> ```python
> paramter_docs = {
> ...
>   10: {
>     "name": _("Vorlauftemperatur"),
>     "description": _("bla bla"),
>     "min": 10.0,
>     "max": 50.0
>   },
> ...
> }
> ```

Can we add a unique textual `entity_key` (e.g. flow_in_temperature) and a not mandatory `visibility_id`?
A part of the params and calulations have a `visibility_id` which says available or not.

**Comment by Bouni** [comment]
@BenPru You mean like the example you posted here: https://github.com/Bouni/python-luxtronik/pull/41#issuecomment-1383218416 ?

**Comment by BenPru** [comment]
> @BenPru You mean like the example you posted here: [#41 (comment)](https://github.com/Bouni/python-luxtronik/pull/41#issuecomment-1383218416) ?

Yes

**Comment by Bouni** [comment]
Was this just a made up example or did you actually start to implement and test this somewhere?

**Comment by BenPru** [comment]
> Was this just a made up example or did you actually start to implement and test this somewhere?

Just example and perhaps to direct use for @kbabioch 😃.

**Comment by Bouni** [comment]
@BenPru You mean having the translations in this core module and for example pass a language code to the Luxtronik calls and then it returns the already translated names?

That sounds good to me!

**Comment by BenPru** [comment]
> the translations in this core module and for example pass a language code to the Luxtronik calls and then it returns the already translated names?

Yes, exact. Perhaps we can add the original ait Luxtronik name. E.g. "AV-Abtauventil":
![image](https://user-images.githubusercontent.com/32298537/215551489-48e8b093-eb82-4d52-b664-2f68f609897e.png)

**Comment by gerw** [comment]
Another possibility would be to put the descriptions of the parameters into a json file.

**Comment by kbabioch** [comment]
> Another possibility would be to put the descriptions of the parameters into a json file.

I like this idea very much. Don't care too much about the exact format of this file (JSON, YAML, Python, etc.). During initialization that file can then be read and used by the library. This file could then be easily edited / translated by other means and it is rather easy to update it without touching the code.

What kind of data do we need to store / manage for each parameter / visibility / calculations?

**Comment by Guzz-T** [comment]
I would assume this issue has been resolved with the current implementation. Can we close it?
